www/js/diary/infinite_scroll_list.js

```diff
@@ -184,76 +249,62 @@ angular.module('emission.main.diary.infscrolllist',['ui-leaflet',
-        Timeline.setInfScrollCompositeTripList($scope.data.allTrips);
-        const oldestTrip = ctList[0];
-        if (oldestTrip) {
-            if (oldestTrip.start_ts <= $scope.infScrollControl.pipelineRange.start_ts) {
-                Logger.log("Oldest trip in batch starts at "+ moment(oldestTrip.start_ts)
-                    +" pipeline starts at "+moment($scope.infScrollControl.pipelineRange.start_ts)
-                    +" reached end");
-                $scope.infScrollControl.reachedEnd = true;
-            } else {
-                // Since this was reversed, the first entry is the most recent
-                $scope.infScrollControl.currentEnd =
-                    oldestTrip.end_ts - 1;
-                Logger.log("new end time = "+$scope.infScrollControl.currentEnd);
-            }
+        if (direction == 'future') {
+          $scope.data.allTrips = $scope.data.allTrips.concat(tripsRead);
         } else {
-            Logger.log("current batch of size 0 but haven't reached pipeline start, going on");
-            $scope.infScrollControl.currentEnd = $scope.infScrollControl.currentEnd - ONE_WEEK;
+          $scope.data.allTrips = tripsRead.concat($scope.data.allTrips);
         }
-        $scope.recomputeDisplayTimelineEntries();
+        Logger.log("After adding batch of size "+tripsRead.length+" cumulative size = "+$scope.data.allTrips.length);
+        Timeline.setInfScrollCompositeTripList($scope.data.allTrips);
+        const oldestTrip = tripsRead[0];
+        const [latestTrip] = tripsRead.slice(-1);
+        $scope.updateTimeRange(direction, oldestTrip, latestTrip);
+        $scope.recomputeListEntries();
         Logger.log("Broadcasting infinite scroll complete");
         $ionicLoading.hide();
         $scope.$broadcast('scroll.infiniteScrollComplete');
 
     }).catch((err) => {
         Logger.displayError("while reading confirmed trips", err);
         Logger.log("Reached the end of the scrolling");
-        $scope.infScrollControl.reachedEnd = true;
+        $scope.infScrollControl.reachedPipelineStart = true;
         Logger.log("Broadcasting infinite scroll complete");
         $ionicLoading.hide();
         $scope.$broadcast('scroll.infiniteScrollComplete');
     });
+    Timeline.readUnprocessedTrips(startTs, endTs, $scope.data.allTrips).then((utList) => {
+      Logger.log("Received unprocessed trip batch of size " + utList.length);
+    }).catch((err) => {
+      Logger.displayError("while reading unconfirmed trips", err);
+    });
   };
 
   $scope.setupInfScroll = function() {
     Logger.log("Setting up the scrolling");
     $scope.itemHt = DEFAULT_ITEM_HT;
-    $scope.infScrollControl.reachedEnd = false;
+    $scope.infScrollControl.reachedPipelineStart = false;
+    $scope.infScrollControl.reachedPipelineEnd = true;
     $scope.data.allTrips = [];
-    $scope.data.displayTimelineEntries = [];
+    $scope.data.listEntries = [];
```

Comment:
> Future fix/refactor: we may want to treat `allTrips` as a fixed length buffer instead of an infinitely growable list. Right now, we only reset `allTrips` here, not when we load another batch of data using "load more"
> 
> That leads to the issue that if a user starts at either end and keeps loading more and more trips, then we could end up with hundreds of entries locally, which can be a memory issue, particularly since the objects are not super small.
> 
> However, the issue exists in the codebase today so it is clearly not a critical fix.

I suggest we re-evaluate this once the list itself has been converted. I've noticed lag while scrolling a lot of trips, especially on my own Pixel 5. I think that Ionic's method of rendering large lists is just not very good, and I think that we will have better luck in React.
Even with several months in the list, which could amount to a couple hundred trips, I don't think memory should be the issue. we've taken some effort to ensure trip objects aren't excessively large.
I think the lagging is just from inefficient rendering; trying to render too many trips at once, instead of just the ones that are visible. I think there are smarter modern solutions for large lists that we'll find in the React ecosystem.
At that point, if we still see performance issues then yes; let's implement the buffer and cap it to some number of trips (based on how much it seems to be able to safely handle)

Reply:
> Additional future cleanup: we may want to rename the variables in `updateFilterSel`. There's a `$scope.allTrips` there which can easily be confused with this `$scope.data.allTrips`. Again, this is an issue in the existing codebase.

This can be remedied sooner rather than later, though

Reply:
> Additional future cleanup: we may want to rename the variables in `updateFilterSel`. There's a `$scope.allTrips` there which can easily be confused with this `$scope.data.allTrips`. Again, this is an issue in the existing codebase.

addressed in #980 - `$scope.allTrips` was not actually being used anywhere so I just removed it altogether.